lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -982,6 +981,17 @@ def unsupported_failure_response(initial_response)
       def card_not_stored?(response)
         response.authorization ? response.authorization.split('#')[2].nil? : true
       end
+
+      def skip_mpi_data?(options = {})
+        # Skips adding the NT mpi data if it is explicitly skipped in options, or if it is MIT and not the initial transaction.
+        options[:skip_mpi_data] == 'Y' || (!options.dig(:stored_credential, :initial_transaction) && options.dig(:stored_credential, :initiator) == 'merchant' && options[:switch_cryptogram_mapping_nt])
```

Comment:
Is this just because you want to test that it works all payment methods types? Because switch_cryptogram_mapping_nt is passed for all payment methods not just NT.

Reply:
Good question, let me look at their docs and make sure I'm not missing something with other payment method types when it comes to mpi_data. I'll ensure that is in a good place before I move this closer to deployment.

Reply:
So the reason I am using the flag here is because I don't want to update the fields we are sending until we activate the flag. `(!options.dig(:stored_credential, :initial_transaction) && options.dig(:stored_credential, :initiator) == 'merchant'` <== this portion is because Adyen's docs tell us to skip it when it is a follow-up merchant initiated transaction. I am adding the flag there because I don't want to start skipping the mpiData in new situations until we full activate this flow.